src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -255,13 +244,39 @@ private async Task<int> PromoteBuildAsync(Build build, List<Channel> targetChann
                 return Constants.ErrorCode;
             }
 
+            // Construct the templateParameters and queue time variables. 
+            // Publishing v2 uses variables and v3 uses parameters, so just use the same values for both.
+            var promotionPipelineVariables = new Dictionary<string, string>
+            {
+                { "BarBuildId", build.Id.ToString() },
+                { "PublishingInfraVersion", _options.PublishingInfraVersion.ToString() },
+                { "PromoteToChannelIds", string.Join(",", targetChannels.Select(tch => tch.Id)) },
+                { "EnableSigningValidation", _options.DoSigningValidation.ToString() },
+                { "SigningValidationAdditionalParameters", _options.SigningValidationAdditionalParameters },
+                { "EnableNugetValidation", _options.DoNuGetValidation.ToString() },
+                { "EnableSourceLinkValidation", _options.DoSourcelinkValidation.ToString() },
+                { "PublishInstallersAndChecksums", _options.PublishInstallersAndChecksums.ToString() },
+                { "SymbolPublishingAdditionalParameters", _options.SymbolPublishingAdditionalParameters },
+                { "ArtifactsPublishingAdditionalParameters", _options.ArtifactPublishingAdditionalParameters }
+            };
+
+            if (_options.DoSDLValidation)
+            {
+                promotionPipelineVariables.Add("EnableSDLValidation", _options.DoSDLValidation.ToString());
+                promotionPipelineVariables.Add("SDLValidationCustomParams", _options.SDLValidationParams);
+                promotionPipelineVariables.Add("SDLValidationContinueOnError", _options.SDLValidationContinueOnError);
+            }
+
+            // Pass the same values to the variables and pipeline parameters so this works with the
+            // v2 and v3 versions of the promotion pipeline.
             int azdoBuildId = await azdoClient.StartNewBuildAsync(build.AzureDevOpsAccount,
                 promotionPipelineInformation.project,
                 promotionPipelineInformation.pipelineId,
                 arcadeSDKSourceBranch,
                 arcadeSDKSourceSHA,
-                queueTimeVariables)
```

Comment:
If we're always passing the same values for both params should one just be eliminated?

Reply:
It's the case for the promotion pipeline, (we're moving from variables to template parameters) they should be the same, but not for all build definitions. In current usage this method is only used for the promotion pipeline, but I need them to not always be coupled for other usages of this function I have in my current work.

Reply:
ahh, there are other usages. Got it...carry on.